code/datums/components/infective.dm

```diff
@@ -1,12 +1,78 @@
 /datum/component/infective
+	dupe_mode = COMPONENT_DUPE_ALLOWED
 	var/list/datum/disease/diseases //make sure these are the static, non-processing versions!
+	var/expire_time
+	var/min_clean_strength = CLEAN_WEAK
 
-/datum/component/infective/Initialize(list/datum/disease/_diseases)
-	RegisterSignal(COMSIG_MOVABLE_CROSSED, .proc/Infect)
-	diseases = _diseases
+/datum/component/infective/Initialize(list/datum/disease/_diseases, expire_in)
+	if(islist(_diseases))
+		diseases = diseases
+	else
+		diseases = list(_diseases)
+	if(expire_in)
+		expire_time = world.time + expire_in
+		QDEL_IN(src, expire_in)
+	RegisterSignal(COMSIG_MOVABLE_BUCKLE, .proc/try_infect_buckle)
+	RegisterSignal(COMSIG_MOVABLE_COLLIDE, .proc/try_infect_collide)
+	RegisterSignal(COMSIG_MOVABLE_CROSSED, .proc/try_infect_crossed)
+	RegisterSignal(COMSIG_ITEM_ATTACK_ZONE, .proc/try_infect_attack_zone)
+	RegisterSignal(COMSIG_ITEM_ATTACK, .proc/try_infect_attack)
+	RegisterSignal(COMSIG_ITEM_EQUIPPED, .proc/try_infect_equipped)
+	RegisterSignal(COMSIG_MOVABLE_IMPACT_ZONE, .proc/try_infect_impact_zone)
+	RegisterSignal(COMSIG_FOOD_EATEN, .proc/try_infect_eat)
+	RegisterSignal(COMSIG_COMPONENT_CLEAN_ACT, .proc/clean)
 
-/datum/component/infective/proc/Infect(atom/movable/AM)
-	var/mob/living/carbon/victim = AM
-	if(istype(victim))
-		for(var/datum/disease/D in diseases)
-			victim.ContactContractDisease(D, "feet")
\ No newline at end of file
+/datum/component/infective/proc/try_infect_eat(mob/living/eater, mob/living/feeder)
+	for(var/V in diseases)
+		eater.ForceContractDisease(V)
+	try_infect(feeder, BODY_ZONE_L_ARM)
```

Comment:
Shouldn't this be something else ? Having covered hands shouldn't protect you if you eat infected food.

Reply:
This line is for the person doing the feeding, the line above this handles the person eating the food.

Reply:
Ahh i misread this.